### ISSUE TITLE
Mupen64Plus support?

### ISSUE THREAD

**Issue by SwooshyCueb** [Azimer/AziAudio#106]
We have a folder in the source named "Mupen64plusHLE" but I don't think we actually support Mupen64Plus? I can get the plugin to work with Mupen64 but Mupen64Plus (or at least m64py) doesn't see it as far as I can tell. I'd totally be willing to work on this but first I'd like to make sure we don't actually have support for it yet.

**Comment by cxd4** [comment]
Probably the Mupen64Plus API is not really supported--some leftover bits of the M64P API are scattered throughout the bsmiles portion of the HLE code to my understanding.

I suppose implementing support for the modified zilmar plugin specs from Mupen64Plus might contribute to open-source activity between more parties, but the Mupen64Plus plugin API really has just all of the flaws and problems as the original zilmar plugin spec, with only a couple fundamental differences that make the two specs incompatible (such as moving all context framework, config system etc. stuff to be controlled by Mupen64Plus instead of the plugin, which they think is more portable when actually it makes the core itself less available to targets which don't support such arbitrary framework elements).  So the modified plugin specs I don't agree with in particular since that kind of attacks its own attempts to improve portability, but I like the idea of having M64P API support anyway if you can figure out a minimalist, small contribution that doesn't interfere much with the current code.

**Comment by SwooshyCueb** [comment]
Yeah that's the idea. I don't want to do anything that would overhaul the plugin. Admittedly I haven't looked too much into the M64P API yet so I still don't know how feasible this is, but I don't think it would be too difficult to figure out.

**Comment by Azimer** [comment]
I have no intention of supporting the M64P API at this time.  I need to focus on polishing PJ64 spec first.

**Comment by bee12341234** [comment]
I'm using the mupin 64 plus on my Samsung Galaxy s6 edge. The emulator runs great. I have an issue with analog support though. When I map my external controllers. The analog sticks won't map properly. My characters move on their own. Please help me with the fix?

**Comment by bee12341234** [comment]
To be exact. When I use analog I move to the left and immediately after I am going right. Do you not support analog. I've tried using my steelseries stratus XL and my 8bitdo snes30 pro which has analog thumb sticks. Both have the same issue

**Comment by cxd4** [comment]
Ask the guys there who forked blight's input plugin and modified it.

Azimer's plugin is for sound output and not for joystick or controller input stuff and is unrelated to M64P.

**Comment by Azimer** [comment]
I decided I will be implementing Mupen64Plus in the future.  Marking this as a feature and todo item.
